File: ice0208/prefix-sum/724.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        left_sum_prefixs = [0]
        for num in nums[:-1]:
            left_sum_prefixs.append(left_sum_prefixs[-1] + num)

        right_sum_prefixs = [0]
        for num in nums[:0:-1]:
            right_sum_prefixs.append(right_sum_prefixs[-1] + num)
        right_sum_prefixs.reverse()

        for i in range(len(nums)):
            if left_sum_prefixs[i] == right_sum_prefixs[i]:
                return i
        return -1


if __name__ == "__main__":
    print(Solution().pivotIndex(nums=[1, 7, 3, 6, 5, 6]))
    print(Solution().pivotIndex(nums=[1, 2, 3]))
    print(Solution().pivotIndex(nums=[2, 1, -1]))

File: moonpoison/prefix_sum/724.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int pivotIndex(vector<int>& nums) 
    {
        int p1=0, p2=0;
        for(int i=1;i<nums.size();++i)
            p2+=nums[i];
        int ind=0;
        while(ind+1!=nums.size())
        {
            if(p1==p2) return ind;
            p1+=nums[ind];
            p2-=nums[ind+1];
            ind++;
        }
        if(p1==p2) return ind;
        return -1;
    }
};